master/buildbot/reporters/telegram.py

```diff
@@ -0,0 +1,1055 @@
+
+    def query_button(self, caption, payload):
+        if isinstance(payload, str) and len(payload) < 64:
+            return {'text': caption, 'callback_data': payload}
+        key = hash(repr(payload))
+        while True:
+            cached = self.bot.query_cache.get(key)
+            if cached is None:
+                self.bot.query_cache[key] = payload
+                break
+            elif cached == payload:
+                break
+            key += 1
+        return {'text': caption, 'callback_data': key}
+
+    @defer.inlineCallbacks
+    def command_START(self, args, **kwargs):
+        yield self.command_HELLO(args)
+        self.bot.reactor.callLater(0.2, self.command_HELP, '')
+
+    def command_NAY(self, args, tmessage, **kwargs):
+        """forget the current command"""
+        replied_message = tmessage.get('reply_to_message')
+        if replied_message:
+            if 'reply_markup' in replied_message:
+                self.bot.edit_keyboard(self.channel.id,
+                                       replied_message['message_id'])
+        if self.userid == self.channel.id:
+            self.send("Never mind...")
+        else:
+            self.send("Never mind, {}...".format(self.user_name))
+    command_NAY.usage = "nay - never mind the command we are currently discussing"
+
+    @Contact.overrideCommand
+    def command_COMMANDS(self, args, **kwargs):
+        if args.lower() == 'botfather':
+            lp = len(self.bot.commandPrefix)
+            commands = self.build_commands()
+            response = []
+            for command in commands:
+                command = command[lp:]
+                if command == 'start':
+                    continue
+                meth = self.getCommandMethod(command, True)
+                doc = getattr(meth, '__doc__', None)
+                if not doc:
+                    doc = command
+                response.append("{} - {}".format(command, doc))
+            if response:
+                self.send('\n'.join(response))
+        else:
+            return super().command_COMMANDS(args)
+
+    @defer.inlineCallbacks
+    @Contact.overrideCommand
+    def command_DANCE(self, args, **kwargs):
+        chat = self.channel['id']
+        msg = yield self.send("**<(^.^<)**")
+        if msg is not None:
+            mid = msg['message_id']
+            self.bot.reactor.callLater(1.0, self.bot.edit_message, chat, mid, "**<(^.^)>**")
+            self.bot.reactor.callLater(2.0, self.bot.edit_message, chat, mid, "**(>^.^)>**")
+            self.bot.reactor.callLater(2.5, self.bot.edit_message, chat, mid, "**(7^.^)7**")
+            self.bot.reactor.callLater(4.0, self.bot.edit_message, chat, mid, "**(>^.^<)**")
+            self.bot.reactor.callLater(5.0, self.bot.delete_message, chat, mid)
+            self.bot.reactor.callLater(5.5, self.bot.send_sticker, chat, random.choice((
+                'CAADAgAD9wEAAsoDBgtCnbBFfI8M_BYE',
```

Comment:
Needs docs of what these stickers are.

Reply:
It is a Telegram stickers hash ID, not base64 of anything special. I have removed the stickers anyway.